### ISSUE TITLE
Enabled vs Enabled For Site is not very intuitive

### ISSUE THREAD

**Issue by codyjames** [craftcms/cms#2899]
### Description
In Craft 3, the "Enabled" vs "Enabled for Site" settings on an entry are quite confusing and not very intuitive.

Here are some examples:

1. When editing a single entry, if setting "Enabled" to false and "Enabled for Site" to true I don't have a clear idea of what the outcome would be.
2. In the entry list view, the "Set Status" bulk update dropdown just says "Enable" and "Disable" and I'm unsure which setting that would be toggling. It's also unclear if that is scoped to the current Site or not.

### Additional info

- Craft version: 3.0.7

**Comment by brandonkelly** [comment]
Agree, that should be made more clear.  Thanks for pointing it out!

**Comment by codyjames** [comment]
This bit me again just now. Is this not an issue others are having? Am I just not understanding something that others are?

**Comment by codyjames** [comment]
Ooops!

**Comment by brandonkelly** [comment]
Sorry I thought you were asking rhetorical questions in the original post.

- **Enabled** is the global status of the entry. If this is disabled, the entry won’t be accessible anywhere. When you change this value, it is changed across all sites.
- **Enabled for site** is the site-specific status of the entry. If this is disabled, the entry won’t be accessible from the currently-selected site only, but will still be accessible from other sites (assuming other sites still have it enabled).

**Comment by codyjames** [comment]
Nope, not your fault. My original post wasn't asking any specific questions, just pointing out some confusing wording/behavior.

My current thinking is this should go back to the previous UI which I think showed the status of every site. I assume there were reasons for moving away from that though.

**Comment by codyjames** [comment]
Ran into this again. This is really not intuitive. Folks think they are "Disabling for site" when they are actually globally bulk-disabling entries. This is not a small issue.

**Comment by brandonkelly** [comment]
Just made some changes for the next release that should reduce the confusion.

First, on the Entries index page, “Disabled” has been relabeled to “Disabled Globally”, and a new “Disabled for Site” option has been added.

<img width="421" alt="screen shot 2018-11-03 at 4 25 24 am" src="https://user-images.githubusercontent.com/47792/47951641-a7163c80-df20-11e8-85af-2cd6024cc2ff.png">

Second, within Edit Entry pages, “Enabled” has been relabeled to “Enabled Globally”, and “Enabled for site” has been relabeled to include the site name.

<img width="348" alt="screen shot 2018-11-03 at 4 27 11 am" src="https://user-images.githubusercontent.com/47792/47951653-df1d7f80-df20-11e8-9f08-6880bcfbba91.png">

And if “Enabled Globally” is unchecked, then the site-specific status doesn’t even show up.

<img width="343" alt="screen shot 2018-11-03 at 4 28 02 am" src="https://user-images.githubusercontent.com/47792/47951661-ef355f00-df20-11e8-9cbf-e0a6cfb06af5.png">

**Comment by bertheyman** [comment]
I see all of the above changes (really useful!), but can't see the global / site specific disable option on the overview. Just the regular enable / disable. Am I missing something here @brandonkelly ?

Using Craft 3.1.30 (latest edition)

**Comment by brandonkelly** [comment]
@bertheyman The “Disabled for Site” option will only show up if a multi-site section is selected.

**Comment by bertheyman** [comment]
Oh I see, thanks for the clarification! 
I was looking at a multi site single, which confused me.

**Comment by amityweb** [comment]
How does one "Enabled for Site" only now? So the entry is only enabled for this site and then all other sites its disabled (unless they have it selected)? 

If I disable "Enabled Globally" the setting for "Enabled for Site" scrolls up and disappears, and the entry is not accessible on the front end anymore. So I cannot diasable "Enabled Globally" but keep "Enabled for Site"

At the moment it looks like I have to "Enabled Globally" then disable "Enabled for Site" on all the sites I dont want it on. So say its 15 sites, we create an entry, then edit 14 other pages to disable it? 

Sorry if I misunderstand, but thats how it looks to me. If this is the case, how does on "Enabled for Site" only?

**Comment by brandonkelly** [comment]
> At the moment it looks like I have to "Enabled Globally" then disable "Enabled for Site" on all the sites I dont want it on. So say its 15 sites, we create an entry, then edit 14 other pages to disable it?

That’s correct. But I can see how the “Enabled Globally” wording can be confusing now. Will have to think of a better way to phrase that.

**Comment by amityweb** [comment]
OK but any reason why its backwards, its a bit of a long winded process to have an entry on a single site. I have not ever come across a situation where you may have a page on all sites except one. Its quite common on multi-language sites that a single country would have a page just for that country. 

What are the options in the meantime? I was going to create a separate section which is saving entries only in that site, but I notice Commerce Products does not have this option. Its not only pages that need to be individual to just a single country, but we would have some products only sold in a single country.

**Comment by brandonkelly** [comment]
So to be clear, for an entry to be enabled somewhere, it has to be both Enabled Globally *and* Enabled for <site name>. In the above screenshot, that entry isn’t going to show up on any sites, until “Enabled Globally” becomes ticked on. Then and only then, it will start showing up for whichever sites have the site-specific status ticked on as well.

It works this way because it does. There’s not good reason it can’t be improved. And we will improve it.

**Comment by engram-design** [comment]
@brandonkelly Might only be a slight niggling thing, but when an entry has “Enabled Globally” disabled, the site dropdown shows all entries as being enabled. I suppose they technically are, but its also confusing for users because it doesn't convey the correct status. You'd have to go through each entry and set `enabledForSite` to be false.

**Comment by brandonkelly** [comment]
@engram-design thanks for pointing that out, fixed for the next release (7fc874fac910b3ef47cd192c6993a6269a742de3).

**Comment by brandonkelly** [comment]
I’ve just made some big improvements to multi-site entry status management for Craft 3.4. I realized that for multi-site entries, there’s really no point in managing the entry’s “global” status, besides as a minor UI convenience when you need to batch-enable/disable an entry everywhere, which was probably not worth all the added confusion. (And in hindsight (it’s 2020 after all), “Enabled globally” sure does seem like something that would take precedence over the “Enabled for site” setting.) And then there’s the fact that it gave users a way to disable/enable entries on sites that they didn’t have permission to view in the first place (#2817) 😱

So here’s what’s changed…

First, the “Set Status” action no longer lists separate “Disabled Globally” and “Disabled for Site” options for multi-site sections; just “Disabled”. When “Disabled” is selected, all single-site entries will get disabled globally, and all multi-site entries will get disabled _only for the current site_.

<img width="127" alt="An expanded “Set Status” element action menu with “Enabled” and “Disabled” options" src="https://user-images.githubusercontent.com/47792/72206861-7e080a80-3447-11ea-8883-0219af57dcbf.png">

Second, the status settings for multi-site entries has been redesigned. Now only the current site will be shown by default:

<img width="378" alt="An entry’s status setting, showing that the entry is enabled for “Site A”" src="https://user-images.githubusercontent.com/47792/72206886-d2ab8580-3447-11ea-9925-38a6931ce4a2.png">

If you click that <img width="21" alt="ellipsis button" src="https://user-images.githubusercontent.com/47792/72207029-7ba6b000-3449-11ea-87e2-715b0ba118a5.png"> button, it will expand to show all  the sites your user account has permission to edit:

<img width="375" alt="An entry’s status setting, showing that the entry is enabled for “Site” and “Site B”" src="https://user-images.githubusercontent.com/47792/72207096-1e5f2e80-344a-11ea-9682-ef25d5ce4428.png">

From there you can toggle the entry’s status across each of those sites individually, or batch set them using the “Enabled everywhere” lightswitch. (Note that is _not_ a direct replacement of the old “Enabled Globally” setting though; just a convenience for batch-enabling/disabling the entry across all of the editable sites.)

**Comment by brandonkelly** [comment]
Here’s a gif of the status setting in action:

<img width="375" alt="An animated GIF of an entry’s status settings being interacted with" src="https://user-images.githubusercontent.com/47792/72207360-443a0280-344d-11ea-88fa-03d05004e30c.gif">

**Comment by engram-design** [comment]
This seems pretty neat (and nice one on the half-on lightswitch). The biggest gripe with multi-site authoring is the fact users have to create the entry, which is enabled in all sites (or controlled by the sections default status), after which they need to switch to each site and disable. They'd much prefer managing it's status across all sites from on central, and immediate location, as you've done here.

This seems like a much better solution - I'm guessing reserved for 3.4?

**Comment by amityweb** [comment]
Absolutely brilliant. Thanks a lot.
Would it be on Commerce also for products? Sometimes my retail customers sell a product in one country not others for example.  Although its a great move to have it for entries for now, thanks a lot.

**Comment by brandonkelly** [comment]
@amityweb We do plan to bring it to Commerce, but probably not in time for 3.0.0, which is planned for the end of the month.

**Comment by vnali** [comment]
this new improvement is excellent @brandonkelly  
but i have two questions
- i noticed that when a section has multiple sites, enable column value in element table is always 1, even if entry is disabled for every sites. am i right? is that ok?
- for a custom element type, i tried to add this new functionality to my element edit page. i copied new added lines from _layouts/element.html to my edit.twig, and at the end i assigned settings to Craft.drafteditor -i don't need draft and revision for my element type though-.
it seems if i copy enabledForSiteValue() function from BaseEntriesController.php to my controller i can have this functionality for my custom element type.
my question is does this functionality planned to be reuseable or is it written only for entries support?

**Comment by brandonkelly** [comment]
> i noticed that when a section has multiple sites, enable column value in element table is always 1, even if entry is disabled for every sites. am i right? is that ok?

Correct, and yes it’s ok, as we’re basically delegating entry enabled status to the site-specific status now.

> does this functionality planned to be reuseable or is it written only for entries support?

Yeah it can be reusable. Main thing is that you must tell the element what all of its site-specific statuses should be when setting `enabledForSite`.

```php
// Only enables/disables the element for the current site:
$element->enabledForSite = true;

// Enables/disables the element across multiple sites  by their IDs:
$element->enabledForSite = [
    1 => true,
    2 => false,
];
```

**Comment by ArtDepartmentMJ** [comment]
@brandonkelly is there any follow-up on this for Commerce?

**Comment by santanu-adak-2019** [comment]
> Just made some changes for the next release that should reduce the confusion.
> 
> First, on the Entries index page, “Disabled” has been relabeled to “Disabled Globally”, and a new “Disabled for Site” option has been added.
> 
> <img alt="screen shot 2018-11-03 at 4 25 24 am" width="421" src="https://user-images.githubusercontent.com/47792/47951641-a7163c80-df20-11e8-85af-2cd6024cc2ff.png">
> 
> Second, within Edit Entry pages, “Enabled” has been relabeled to “Enabled Globally”, and “Enabled for site” has been relabeled to include the site name.
> 
> <img alt="screen shot 2018-11-03 at 4 27 11 am" width="348" src="https://user-images.githubusercontent.com/47792/47951653-df1d7f80-df20-11e8-9f08-6880bcfbba91.png">
> 
> And if “Enabled Globally” is unchecked, then the site-specific status doesn’t even show up.
> 
> <img alt="screen shot 2018-11-03 at 4 28 02 am" width="343" src="https://user-images.githubusercontent.com/47792/47951661-ef355f00-df20-11e8-9cbf-e0a6cfb06af5.png">

I want to add "**Enabled Globally**" option in each entry page?
I have posted on this topic Can you please explain me, See the link below
[https://craftcms.stackexchange.com/questions/38640/how-to-add-enabled-globally-in-each-entry-page](url)

**Comment by brandonkelly** [comment]
@santanu-adak-2019 I responded to your Stack Exchange post.
